### ISSUE TITLE
Frontend won't build in development (and in production)

### ISSUE THREAD

**Issue by tmorris42** [fyusuf-a/ft_transcendence#170]
# To reproduce
### Env
```
# Build values
NODE_IMAGE=lts-alpine
NGINX_IMAGE=stable-alpine
BACKEND_DOCKERFILE=Dockerfile
FRONTEND_DOCKERFILE=Dockerfile
```
### Command
`docker-compose --profile debug --profile frontend up`
### Error
```
Build finish
ERROR in src/dtos/matches/match.dto.ts:3:22
ft_transcendence-frontend-1  | TS7016: Could not find a declaration file for module 'class-transformer'. '/app/node_modules/class-transformer/cjs/index.js' implicitly has an 'any' type.
ft_transcendence-frontend-1  |   Try `npm i --save-dev @types/class-transformer` if it exists or add a new declaration (.d.ts) file containing `declare module 'class-transformer';`
ft_transcendence-frontend-1  |     1 | import { ApiProperty } from '@nestjs/swagger';
ft_transcendence-frontend-1  |     2 | import { IsDate, IsInt, IsPositive, IsEnum } from 'class-validator';
ft_transcendence-frontend-1  |   > 3 | import { Type } from 'class-transformer';
ft_transcendence-frontend-1  |       |                      ^^^^^^^^^^^^^^^^^^^
ft_transcendence-frontend-1  |     4 |
ft_transcendence-frontend-1  |     5 | export enum MatchStatusType {
ft_transcendence-frontend-1  |     6 |   HOME = 'HOME',
ft_transcendence-frontend-1  | 
ft_transcendence-frontend-1  | ERROR in src/dtos/pages/page-options.dto.ts:4:22
ft_transcendence-frontend-1  | TS7016: Could not find a declaration file for module 'class-transformer'. '/app/node_modules/class-transformer/cjs/index.js' implicitly has an 'any' type.
ft_transcendence-frontend-1  |   Try `npm i --save-dev @types/class-transformer` if it exists or add a new declaration (.d.ts) file containing `declare module 'class-transformer';`
ft_transcendence-frontend-1  |     2 |
ft_transcendence-frontend-1  |     3 | import { ApiPropertyOptional } from '@nestjs/swagger';
ft_transcendence-frontend-1  |   > 4 | import { Type } from 'class-transformer';
ft_transcendence-frontend-1  |       |                      ^^^^^^^^^^^^^^^^^^^
ft_transcendence-frontend-1  |     5 | import { Min, Max, IsEnum, IsInt, IsOptional } from 'class-validator';
ft_transcendence-frontend-1  |     6 |
ft_transcendence-frontend-1  |     7 | export enum Order {
ft_transcendence-frontend-1  | 
```

### Notes
- running `npm i --save-dev @types/class-transformer` as suggested gives the error that `'@types/class-transformer@*' is not in this registry.`
- #166 does not seem to affect this error

**Comment by tmorris42** [comment]
# Possible Solution
Create a file `class-transformer.d.ts` which contains:
```
declare module 'class-transformer';
```

This file could do in a special `declarations` directory, or in the dto directory.

**Comment by tmorris42** [comment]
Seems to be fixed by #166 after all
